src/mono/mono/metadata/custom-attrs.c

```diff
@@ -719,11 +723,11 @@ load_cattr_value_noalloc (MonoImage *image, MonoType *t, const char *p, const ch
 		return result;
 	}
 	case MONO_TYPE_VALUETYPE:
-		if (m_class_is_enumtype (t->data.klass)) {
-			type = mono_class_enum_basetype_internal (t->data.klass)->type;
+		if (m_class_is_enumtype (m_type_data_get_klass (t))) {
```

Comment:
Could we at least put m_type_data_get_klass (t) into a local only doing call once for this block?